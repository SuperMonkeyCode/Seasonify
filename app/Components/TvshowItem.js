import { Body, ListItem, Text } from 'native-base';
import React from 'react';

export default function TvshowItem({
  tvshowId,
  tvshowName,
  navigate,
}: {
  tvshowId: string,
  tvshowName: string,
  navigate: Function,
}) {
  const onPressTvshow = () => {
    navigate('TvshowDetailsPage', {
      tvshowId,
      tvshowName,
    });
  };

  return (
    <ListItem onPress={onPressTvshow}>
      <Body>
        <Text>{tvshowName}</Text>
      </Body>
    </ListItem>
  );
}
