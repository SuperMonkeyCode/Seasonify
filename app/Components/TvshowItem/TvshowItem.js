/* @flow */
import { Badge, Body, Button, Left, ListItem, Text, Thumbnail, View } from 'native-base';
import { compose, pure, withHandlers } from 'recompose';
import React from 'react';

import styles from './TvshowItem.style';

type Props = {
  /* parent */
  tvshowId: string,
  navigate: Function,
  /* connect */
  tvshow: Tvshow,
  seasonsCount: number,
  viewers: ViewerInfo[],
  /* HOC */
  handlePress: Function,
};

const enhance = compose(
  pure,
  withHandlers({
    handlePress: ({ tvshowId, navigate }: Props) => () => {
      navigate('TvshowDetailsPage', { tvshowId });
    },
  }),
);

const renderViewer = (viewer: ViewerInfo, seasonsCount: number) => {
  const whenNotCompleted =
    viewer.seasonsViewed < seasonsCount ? styles.viewerNotCompleted : {};
  return (
    <Text note numberOfLines={1} key={viewer.name} style={styles.viewer}>
      {viewer.name}{' '}
      <Text note style={whenNotCompleted}>
        {viewer.seasonsViewed}
      </Text>&nbsp;&nbsp;
    </Text>
  );
};
function TvshowItem({ tvshow, seasonsCount, viewers, handlePress }: Props) {
  return (
    <ListItem avatar onPress={handlePress}>
      <Left>
        {tvshow.poster &&
          <Thumbnail square source={{ uri: tvshow.poster }} style={styles.poster} />}
        {!tvshow.poster &&
          <Button disabled transparent style={{ width: 57 }} /> /* FIXME: too tricky */}
      </Left>
      <Body style={styles.infos}>
        <Text style={styles.name}>
          {tvshow.name}
        </Text>
        <Badge style={styles.seasonsCount}>
          <Text>
            {seasonsCount}
          </Text>
        </Badge>
        <View style={styles.viewers}>
          {viewers.length
            ? viewers.map(viewer => renderViewer(viewer, seasonsCount))
            : <Text>&nbsp;</Text>}
        </View>
      </Body>
    </ListItem>
  );
}

export default enhance(TvshowItem);
