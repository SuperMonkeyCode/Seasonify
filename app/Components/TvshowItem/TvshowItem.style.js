export default {
  seasonsCount: {
    fontStyle: 'italic',
    color: 'gray',
  },
};
