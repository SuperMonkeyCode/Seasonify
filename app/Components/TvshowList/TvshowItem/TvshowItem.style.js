export default {
  poster: {},
  infos: {
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-between',
    alignItems: 'stretch',
    paddingRight: 7,
  },
  name: {
    flexGrow: 1,
  },
  seasonsCount: {
    backgroundColor: 'lightgray',
  },
  viewers: {
    flexBasis: '100%',
    flexDirection: 'row',
    flexWrap: 'wrap',
  },
  viewer: {},
  viewerNotCompleted: {
    color: 'gray',
    fontWeight: 'bold',
  },
};
