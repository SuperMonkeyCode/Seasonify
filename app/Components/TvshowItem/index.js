/* @flow */
import { connect } from 'react-redux';

import { tvshowSelectors } from 'State/tvshowState';

import TvshowItem from './TvshowItem';

export default connect(
  () => (state, { tvshowId }) => ({
    seasonsCount: tvshowSelectors.makeGetSeasonsCount()(state, { tvshowId }),
    tvshow: tvshowSelectors.getTvshow(state, { tvshowId }),
  }),
  {},
)(TvshowItem);
