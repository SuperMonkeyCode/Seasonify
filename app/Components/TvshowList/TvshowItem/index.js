/* @flow */
import { connect } from 'react-redux';

import { tvshowSelectors } from 'Store/tvshowStore';
import { viewingSelectors } from 'Store/viewingStore';

import TvshowItem from './TvshowItem';

export default connect(
  () => (state, { tvshowId }) => ({
    seasonsCount: tvshowSelectors.makeGetSeasonsCount()(state, { tvshowId }),
    tvshow: tvshowSelectors.getTvshow(state, { tvshowId }),
    viewers: viewingSelectors.makeGetViewersArray()(state, { tvshowId }),
  }),
  {},
)(TvshowItem);
