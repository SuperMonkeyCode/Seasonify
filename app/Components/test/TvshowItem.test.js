import React from 'react';
import TvshowItem from '../TvshowItem';

function setup(props = {}) {
  const defaultProps = {
    tvshowId: 'abc',
    tvshowName: 'Deadwood',
    navigate: jest.fn(),
  };
  const component = shallow(<TvshowItem {...defaultProps} {...props} />);

  return {
    component,
  };
}

describe('rendering', () => {
  describe('with a tvshow name', () => {
    it('should match snapshot', () => {
      const { component } = setup();
      expect(component).toMatchSnapshot();
    });
  });
});
