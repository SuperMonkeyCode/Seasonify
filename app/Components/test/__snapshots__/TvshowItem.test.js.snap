// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering with a tvshow name should match snapshot 1`] = `
<Styled(ListItem)
  onPress={[Function]}
>
  <Styled(Text)>
    Deadwood
  </Styled(Text)>
</Styled(ListItem)>
`;
